File: app/src/main/java/com/cliniva/enventory/viewholder/CustomerHolder.java
package com.cliniva.enventory.viewholder;

import android.databinding.ViewDataBinding;
import android.support.annotation.NonNull;

import com.cliniva.enventory.adapter.base.BaseRecyclerViewHolder;
import com.cliniva.enventory.databinding.ItemCustomerBinding;
import com.cliniva.enventory.event.OnCustomerItemClickedListener;
import com.cliniva.enventory.model.Customer;

public class CustomerHolder extends BaseRecyclerViewHolder<Customer, OnCustomerItemClickedListener> {

    private ItemCustomerBinding binding;

    public CustomerHolder(@NonNull ViewDataBinding itemView) {
        super(itemView);
        this.binding = (ItemCustomerBinding) itemView;
    }

    @Override
    public void onBindView(Customer item, int position, OnCustomerItemClickedListener listener) {
        enableItemViewClick(item, listener);
        binding.setCustomer(item);
    }
}

File: app/src/main/java/com/amirsons/inventory/recyclerview/viewholder/CustomerHolder.kt
package com.amirsons.inventory.recyclerview.viewholder

import android.view.View
import com.amirsons.inventory.recyclerview.base.BaseRecyclerViewHolder
import com.amirsons.inventory.event.OnCustomerItemClickedListener
import com.amirsons.inventory.datamanager.model.Customer
import kotlinx.android.synthetic.main.item_customer.view.*

class CustomerHolder(itemView: View) : BaseRecyclerViewHolder<Customer, OnCustomerItemClickedListener>(itemView) {

    override fun onBindView(item: Customer, position: Int, listener: OnCustomerItemClickedListener?) {

        itemView.tv_customer_name.text = item.name
        itemView.tv_item_amount.text = item.receivableAmount.toString()
        itemView.tv_item_invoice_date.text = item.lastInvoiceDate

        enableItemViewClick(item, listener)
    }
}
